pkg/bootstrap/platform/gcp.go

```diff
@@ -177,18 +216,33 @@ func (e *gcpEnv) Metadata() map[string]string {
 		e.metadata = md
 		return md
 	}
-	if in, err := instanceNameFn(); err == nil {
-		md[GCEInstance] = in
-	}
-	if id, err := instanceIDFn(); err == nil {
-		md[GCEInstanceID] = id
-	}
-	if it, err := instanceTemplateFn(); err == nil {
-		md[GCEInstanceTemplate] = it
+
+	suppliers := []metadataSupplier{
+		createMetadataSupplier(GCEInstance, instanceNameFn),
+		createMetadataSupplier(GCEInstanceID, instanceIDFn),
+		createMetadataSupplier(GCEInstanceTemplate, instanceTemplateFn),
+		createMetadataSupplier(GCEInstanceCreatedBy, createdByFn),
 	}
-	if cb, err := createdByFn(); err == nil {
-		md[GCEInstanceCreatedBy] = cb
+
```

Comment:
Seems identical to L192, should we abstract it out?

Reply:
done